### ISSUE TITLE
Templating support for support-bundle and preflight

### ISSUE THREAD

**Issue by xavpaice** [replicatedhq/troubleshoot#1150]
In https://troubleshoot.sh/docs/analyze/cluster-pod-statuses/ we provide an example:

```yaml
          - fail:
              when: "!= Healthy" # Catch all unhealthy pods. A pod is considered healthy if it has a status of Completed, or Running and all of its containers are ready.
              message: Pod {{ .Namespace }}/{{ .Name }} is unhealthy with a status of {{ .Status.Reason }}.

```

This uses `{{ template }}` formatting which is the same for any other usage of Go templating.  If we want to provide a template to, say, Helm, which outputs a valid spec, we want to be able to use templating in the spec (ie. template a template).

This may or may not be a problem.

Example Helm template:

```yaml
{{- define "preflight.spec" }}
apiVersion: troubleshoot.sh/v1beta2
kind: Preflight
metadata:
 name: preflight-tutorial
spec:
  {{ if eq .Values.mariadb.enabled false }}
  collectors:  
    - mysql:
        collectorName: mysql
        uri: '{{ .Values.externalDatabase.user }}:{{ .Values.externalDatabase.password }}@tcp({{ .Values.externalDatabase.host }}:{{ .Values.externalDatabase.port }})/{{ .Values.externalDatabase.database }}?tls=false'
  {{ end }}
  analyzers:
    - nodeResources:
        checkName: Node Count Check
        outcomes:
          - fail:
              when: 'count() > {{ .Values.global.node_count }}'
              message: The cluster has more nodes than the 3 you are licensed for.
          - pass:
              message: The number of nodes matches your license ({{ .Values.global.node_count }})
    - clusterVersion:
        outcomes:
          - fail:
              when: "< 1.16.0"
              message: The application requires at least Kubernetes 1.16.0, and recommends 1.18.0.
              uri: https://kubernetes.io
          - warn:
              when: "< 1.18.0"
              message: Your cluster meets the minimum version of Kubernetes, but we recommend you update to 1.18.0 or later.
              uri: https://kubernetes.io
          - pass:
              message: Your cluster meets the recommended and required versions of Kubernetes.
   {{ if eq .Values.mariadb.enabled false }}
    - mysql:
        checkName: Must be MySQL 8.x or later
        collectorName: mysql
        outcomes:
          - fail:
              when: connected == false
              message: Cannot connect to MySQL server
          - fail:
              when: version < 8.x
              message: The MySQL server must be at least version 8
          - pass:
              message: The MySQL server is ready
    {{ end }}
{{- end }}
```

When this is rendered, everything inside `{{ }}` is rendered, if I have items like `{{ .Status.Reason }}` (i.e. not Helm) that will cause an error.

If I wanted to add an analyzer that uses `{{ .Namespace}}/{{ .Name }}` and so on, how would I do that?

**Comment by CpuID** [comment]
Couple of ideas:

- escape the `{{` / `}}` blocks? `\{\{`
  - con: pretty ugly looking/hard to read

- do we support an alternative delimiter in troubleshoot? eg. https://medium.com/@etiennerouzeaud/change-default-delimiters-templating-with-template-delims-857938a0b661
  - https://pkg.go.dev/text/template#Template.Delims
  - we could probably still support the existing syntax (for non-Helm usage) + a new delimiter in troubleshoot?
  - this feels like it could be reasonably elegant UX wise... would need to be documented well though so people are aware
  - what do we use? `[[` / `]]`? something else?

- I know KOTS does something using `repl{{`, but this feels like it would be difficult to approach as Helm gets to do the first rendering pass, and would catch the string and ignore the `repl` bit...
  - non-starter?

**Comment by xavpaice** [comment]
https://github.com/helm/helm/issues/2798 has some good examples of patterns that can be used.

Next steps for this:  Document the options, and our recommendation, in the [docs](https://troubleshoot.sh/docs/)

tl;dr:

```yaml
message: Pod {{ .Namespace }}/{{ .Name }} is unhealthy with a status of {{ .Status.Reason }}.
```

Becomes:
```yaml
message: {{ `Pod {{ .Namespace }}/{{ .Name }} is unhealthy with a status of {{ .Status.Reason }}.` }}
```

There's other Go template methods, but it seems that ```{{ `some literal string` }}``` is the simple answer

**Comment by CpuID** [comment]
that approach of using ` to render a string literal looks decent 👍
